geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionDataStore.java

```diff
@@ -265,6 +265,36 @@ public void visit(Integer bucketId, Region r) {
     return numPrimaries.get();
   }
 
+  public boolean isPartitionedRegionReady(PartitionedRegion partitionedRegion, final int bucketId) {
+    List<PartitionedRegion> colocatedWithList = getColocatedChildRegions(partitionedRegion);
+    if (colocatedWithList.size() == 0) {
+      return partitionedRegion.isInitialized();
+    }
+    return areAllColocatedPartitionedRegionsReady(bucketId, colocatedWithList);
+  }
+
+  private boolean areAllColocatedPartitionedRegionsReady(int bucketId,
+      List<PartitionedRegion> colocatedWithList) {
+    return !colocatedWithList.stream()
```

Comment:
I'm not sure that the stream and filter use here make the code clearer. It also requires that isColocatedPartitionedRegionInitialized be called for every item in the List when it could quit early on the first one that returns false. The following seems clearer but that might be because I'm more used to it:
for (PartitionedRegion partitionedRegion: colocatedWithList) {
  if (!isColocatedPartitionedRegionInitialized(partitionedRegion, bucketId)) {
    return false;
  }
}
return true;

If you decide the stream stuff is better than I would not use "== false" but instead have it be "!".
Also instead of !findFirst().isPresent() I think "empty()" is clearer. If we don't find any that are not initialized then all are ready.